core/src/main/java/org/mule/runtime/core/api/el/ExpressionManager.java

```diff
@@ -0,0 +1,280 @@
+   * within the expression to be maintained post-evaluation via the use of a result
+   * {@link org.mule.runtime.core.api.Event.Builder} which should be created from the original event before being passed and then
+   * used to construct the post-evaluation event.
+   *
+   * @param expression
+   * @param event
+   * @param eventBuilder
+   * @param flowConstruct
+   * @param object
+   */
+  void enrich(String expression, Event event, Event.Builder eventBuilder, FlowConstruct flowConstruct, Object object);
+
+  /**
+   * Enriches an event using a typed value.
+   *
+   * This version of {@code enrich} allows {@link Event} or {@link InternalMessage} mutation performed
+   * within the expression to be maintained post-evaluation via the use of a result
+   * {@link org.mule.runtime.core.api.Event.Builder} which should be created from the original event before being passed and then
+   * used to construct the post-evaluation event.
+   *
+   * @param expression a single expression i.e. header://foo that defines how the message should be enriched
+   * @param event The event to be enriched
+   * @param eventBuilder event builder instance used to mutate the current message or event.
+   * @param flowConstruct the flow where the event is being processed
+   * @param value The typed value used for enrichment
+   */
+  void enrich(String expression, Event event, Event.Builder eventBuilder, FlowConstruct flowConstruct, TypedValue value);
+
+  boolean evaluateBoolean(String expression, Event event, FlowConstruct flowConstruct) throws ExpressionRuntimeException;
+
+  boolean evaluateBoolean(String expression, Event event, FlowConstruct flowConstruct, boolean nullReturnsTrue,
+                          boolean nonBooleanReturnsTrue)
+      throws ExpressionRuntimeException;
+
+  /**
+   * Evaluates expressions in a given string. This method will iterate through each expression and evaluate it. If a user needs to
+   * evaluate a single expression they can use {@link #evaluate(String, Event, FlowConstruct, BindingContext)}.
+   *
+   * This version of {@code evaluate} performs expression evaulation on an immutable event. Any {@link Event} or
+   * {@link InternalMessage} mutation performed within the expression will impact within the context of
+   * expression evaluation but will not mutated the {@code event} parameter.
+   *
+   * @param expression one or more expressions ebedded in a literal string i.e. "Value is #[xpath://foo] other value is
+   *        #[header:foo]."
+   * @param event The current event being processed
+   * @param flowConstruct the flow where the event is being processed
+   * @return the result of the evaluation
+   * @throws ExpressionRuntimeException if the expression is invalid, or a null is found for the expression and 'failIfNull is set
+   *         to true.
+   */
+  String parse(String expression, Event event, FlowConstruct flowConstruct) throws ExpressionRuntimeException;
```

Comment:
Mule 4.0 should not support "parse" operations. Should this functions be in the extended expression manager?